book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,132 @@
+Is worth mentioning that the result are specific to the code and even the computer running the benchmark and thus not useful for comparisons with other projects.
+
+### Kimchi memory
+
+Similar to time, but with memory, the constraints are different in some ways, noise is not generally a problem and any computer cab be used, and also requires less sampling and should run faster. This is to some degree already implemented through a custom measurement for criterion, and in practice can be slower than measuring time, the instrumentation makes it run slower and the some limitations in criterion require to run the code more than necessary. With that in mind the best would be to run just a subset of these benchmark for regressions and let the others be run at discretion.
+
+### Kimchi memory flamegraph
+
+Similar to the time flamegraph, ideally can be made from the memory benchmark without much additional work.
+
+### Snarky time
+
+This is similar to kimchi time, but the fact that it will be in the mina repo should be considered, CI will run much more often and costs may be higher, this would be a reason to prefer to have benchmarks in kimchi when possible.
+
+### Snarky memory
+
+Here the main interest may be regressions, everything else can probably done at the kimchi level more precisely.
+
+### Pickles time
+
+This is similar to snarky time, mostly differing in the overhead of the recursion. At this level beyond some regression we will probably have mostly application circuits being benchmarked.
+
+### Pickles memory
+
+Similar to snarky, this will probably be mostly regression tests and maybe some example application to give users an idea of what to expect.
+
+## Benchmarks
+
+Here is a non exhaustive list of benchmarks.
+
+### Kimchi hash chain
+
+Benchmarking different aspects of a circuit proving a hash chain.
+
+- The length of the hash chain can easily be a parameter allowing to measure different lengths. 
+- We want memory and time for compilation, proving and verification, it will be implemented as several benchmarks but they will share most code.
+- The number of constraints can be easily shown for each length.
+- This benchmark with some specific length can be used as regression tests in CI.
+- The full benchmark with several lengths can be shown at discretion of those interested running locally, or even added as optional to CI with some way to trigger it.
+
+### Pickles hash chain
+
+Same as kimchi hash chain but in pickles, we have less to cover here because we already have the kimchi ones, some time and memory benchmarks could be used for regressions and to observe the overhead, but having parameters like different circuits sizes may be redundant.
+A new benchmark possible would be to have an alternative circuit that uses recursion to make the hash chain instead of a single circuit, it could be interesting to have a comparison to the non recursive approach.
```

Comment:
```suggestion
A new  possible benchmark would be to have an alternative circuit that uses recursion to make the hash chain instead of a single circuit. It could be interesting to have a comparison to the non recursive approach.
```